digitalocean/datasource_digitalocean_container_registry.go

```diff
@@ -0,0 +1,40 @@
+package digitalocean
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func dataSourceDigitalOceanContainerRegistry() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDigitalOceanContainerRegistryRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "name of the container registry",
+				ValidateFunc: validation.NoZeroValues,
+			},
+		},
+	}
+}
+
+func dataSourceDigitalOceanContainerRegistryRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*CombinedConfig).godoClient()
+
+	reg, response, err := client.Registry.Get(context.Background())
+
+	if err != nil {
```

Comment:
@andrewsomething should we check if the registry name matches?

Reply:
The current implementation in godo assumes that there will always be only one registry per account. Checking for the name might help safeguard us from a future change in behaviour, but that is going to be tricky either way.